modules/identity/src/Volo.Abp.Identity.Domain/Volo/Abp/Identity/Permissions/UserRolePermissionValueProvider.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Volo.Abp.Authorization.Permissions;
+using Volo.Abp.Security.Claims;
+
+namespace Volo.Abp.Identity.Permissions;
+
+public class UserRolePermissionValueProvider : PermissionValueProvider
+{
+    public const string ProviderName = "UR";
+
+    public override string Name => ProviderName;
+
+    protected IUserRoleFinder UserRoleFinder { get; }
+
+    public UserRolePermissionValueProvider(IPermissionStore permissionStore, IUserRoleFinder userRoleFinder)
+        : base(permissionStore)
+    {
+        UserRoleFinder = userRoleFinder;
+    }
+
+    public async override Task<PermissionGrantResult> CheckAsync(PermissionValueCheckContext context)
+    {
+        var userId = context.Principal?.FindFirst(AbpClaimTypes.UserId)?.Value;
+
+        if (userId == null)
+        {
+            return PermissionGrantResult.Undefined;
+        }
+
+        if (!Guid.TryParse(userId, out var userGuidId))
+        {
+            return PermissionGrantResult.Undefined;
+        }
+
+        var userRoles = await UserRoleFinder.GetRoleNamesAsync(userGuidId);
```

Comment:
`UserPermissionValueProvider` will return undefined if current user has no permission. Then `UserRolePermissionValueProvider` (this class) will work and try to find roles with a remote API call (for a microservice scenario). A remote API call is not good for permission check that should work fast.
Notice that it work always, not only when we want to check a user's permission with a temporary principle with userid claim. That's unacceptable.

Reply:
Another problem is that: Now, `Volo.Abp.Identity.Domain` package should be added to every microservice needs to check a user's permission with id.

Reply:
Another problem: When we create a temporary principle only with userid claim, some custom permission value providers may not work properly (if they are checking another claim). So, creating a temporary principle only with userid claim is a partial workaround solution. True implementation should be creating real principle of the target user. However, it requires much more work to be done and should be well-designed by considering possible use cases.